pyat/at/collective/beam_loading.py

```diff
@@ -13,7 +13,10 @@ class BLMode(IntEnum):
     WAKE = 1
     PHASOR = 2
 
-
+class CavityMode(IntEnum):
+    ACTIVE = 1
+    PASSIVE = 2
+    
 def add_beamloading(ring: Lattice, qfactor: Union[float, Sequence[float]],
```

Comment:
Here you have to be careful depending on the mode you want vgen to be zero all the time: in build_from_cav and in init_bl_param you want this to be correctly handled

Unfortunately I can comment only the line that you changed...but lines 298 and 209 need to be adapted depending on the mode

Also do you want to allow setting Vgen for passive cavity? Should you return an error?

Reply:
done